pyuvdata/miriad.py

```diff
@@ -268,9 +317,7 @@ def read_miriad(self, filepath, correct_lat_lon=True, run_check=True,
             data_accumulator[pol] = np.array(data)
 
         self.polarization_array = np.array(pol_list)
-        if len(self.polarization_array) != self.Npols:
-            warnings.warn('npols={npols} but found {n} pols in data file'.format(
-                npols=self.Npols, n=len(self.polarization_array)))
+        self.Npols = len(pol_list)
```

Comment:
I'm a little confused about this, is the Miriad `npol` parameter not available unless you read in the whole file?

Reply:
no it is available, but when you perform a `UV.select`  on the miriad file on polarizations, `npol` doesn't change to reflect the extracted polarizations, it remains what it was for the entire file..

Reply:
I see. Maybe we can leave in the check unless we're doing a select along the polarization axis?

Reply:
well its not so much that the test breaks things, its that with the new definition of `Npols`, its a trivially-true and non-helpful test.. because  I'm manually setting `self.Npols` to be equal to `len(pol_list)`, which itself has explicitly set to `len(self.polarization_array)`, if you see what I mean.

Reply:
Yes, I see. I'm just thinking that in the case where there's no down select on the polarization axis, we could still check to see if npol matched the polarization array length. But it's really not critical, so I think it's ok to drop it if it doesn't make sense.